docs/rules/jsx-props-no-spreading.md

```diff
@@ -0,0 +1,24 @@
+# Disallow JSX props spreading (react/jsx-props-no-spreading)
+
+Enforces that there is no spreading for any JSX attribute. This enhances readability of code and hence leads to better maintainability.
```

Comment:
Let's elaborate here on why it helps maintainability - this should talk about a) avoiding passing unintentional extra props, b) being explicit about what props are received and written, c) avoiding react warnings when invalid HTML props are passed to HTML elements, etc.